sklearn/cluster/tests/test_affinity_propagation.py

```diff
@@ -66,7 +67,14 @@ def test_affinity_propagation_predict():
     labels2 = af.predict(X)
     assert_array_equal(labels, labels2)
 
-
+    
+def test_affinity_propagation_convergence_warning():
+    # Test warning in AffinityPropagation.fit
+    af = AffinityPropagation(affinity="euclidean", max_iter=1)
+    assert_warns(ConvergenceWarning, af.fit, X)
+    assert_raises(ValueError, af.predict, X)
+    
```

Comment:
Is it ok if it requires "magic" parameters? It seems you have to have a particular combination of bad damping and bad preferences combined with the right number of max_iter to get it to not converge. Sometimes it will "converge" but return # clusters = # data ...